### PR TITLE
Status in rest

### PR DESCRIPTION
**Pull request by radix**
implements part 2 of #885.

This makes it so the 'state' object in REST results for creation/getting groups has the 'status' key.

### CODE REVIEW

**Comment by radix** [comment]
jenkins please run integration tests

**Comment by radix** [comment]
jenkins please run unit tests

**Comment by radix** [comment]
jenkins please run integration tests

**Comment by radix** [comment]
jenkins please run integration tests

File: otter/test/rest/test_groups.py [hunk 1]
```diff
@@ -134,6 +135,37 @@ def test_format_state_dict_with_convergence(self, config_value):
...
+            None,
+            {},
+            True,
+            ScalingGroupStatus.ERROR,
+            desired=10)
+        result = format_state_dict(state)
+        self.assertEqual(result['desiredCapacity'], 10)
+        self.assertEqual(result['pendingCapacity'], 7)
+
+        # And a non-convergence tenant still gets old-style data
```
Comment (cyli) [inline otter/test/rest/test_groups.py]: Is the point of this part that a non-convergence tenant also gets the error status?  Should we also test that the convergence tenant gets the status too?

**Comment by cyli** [comment]
Aside from the above, LGTM!

File: otter/test/rest/test_groups.py (hunk 1 above)
Comment (radix) [inline otter/test/rest/test_groups.py]: er, sorry, no, that was mistakenly copied from another unit test :) I'll delete it :)

**Comment by radix** [comment]
thank you @cyli :)

**Pull request merged**